code/modules/projectiles/guns/ballistic/automatic.dm

```diff
@@ -165,7 +165,8 @@
 	update_icon()
 
 /obj/item/gun/ballistic/automatic/m90/afterattack(atom/target, mob/living/user, flag, params)
-	if(select == 2)
+	var/list/modifiers = params2list(params)
+	if(modifiers["right"] && !Adjacent(target))
```

Comment:
Use the new alt attack procs, and don't bother checking adjacency.

Reply:
I was checking adjacency so you could still hold people up with the gun without blowing yourself up, are you sure?

Reply:
After staring at a screen confused for several hours, I've finally found that the alt attack procs aren't even called when a gun is fired. I feel slightly less sane, thanks.

Reply:
> I was checking adjacency so you could still hold people up with the gun without blowing yourself up, are you sure?

The staring has also made me realize why I didn't need to check adjacency.